src/class-post-duplicator.php

```diff
@@ -0,0 +1,360 @@
+			 *
+			 * @return array.
+			 */
+			$new_post = \apply_filters( 'duplicate_post_new_post', $new_post, $post );
+		}
+		$new_post_id = \wp_insert_post( \wp_slash( $new_post ), true );
+
+		if ( ! \is_wp_error( $new_post_id ) ) {
+			\delete_post_meta( $new_post_id, '_dp_original' );
+			\add_post_meta( $new_post_id, '_dp_original', $post->ID );
+		}
+
+		return $new_post_id;
+	}
+
+	/**
+	 * Wrapper function to create a copy for the Rewrite and Republish feature.
+	 *
+	 * @param \WP_Post $post The original post object.
+	 *
+	 * @return number|\WP_Error The copy ID, or a WP_Error object on failure.
+	 */
+	public function create_duplicate_for_rewrite_and_republish( \WP_Post $post ) {
+		$options  = [
+			'copy_title'      => true,
+			'copy_date'       => true,
+			'copy_name'       => true,
+			'copy_content'    => true,
+			'copy_excerpt'    => true,
+			'copy_author'     => true,
+			'copy_menu_order' => true,
+			'use_filters'     => false,
+		];
+		$defaults = $this->get_default_options();
+		$options  = \wp_parse_args( $options, $defaults );
+
+		$new_post_id = $this->create_duplicate( $post, $options );
+
+		if ( ! \is_wp_error( $new_post_id ) ) {
+			$this->copy_post_taxonomies( $new_post_id, $post, $options );
+			$this->copy_post_meta_info( $new_post_id, $post, $options );
+
+			\update_post_meta( $new_post_id, '_dp_is_rewrite_republish_copy', 1 );
+			\update_post_meta( $post->ID, '_dp_has_rewrite_republish_copy', $new_post_id );
+		}
+
+		return $new_post_id;
+	}
+
+	/**
+	 * Copies the taxonomies of a post to another post.
+	 *
+	 * @param int      $new_id  New post ID.
+	 * @param \WP_Post $post    The original post object.
+	 * @param array    $options The options array.
+	 *
+	 * @return void
+	 */
+	private function copy_post_taxonomies( $new_id, $post, $options ) {
+		// Clear default category (added by wp_insert_post).
+		\wp_set_object_terms( $new_id, null, 'category' );
+
+		$post_taxonomies = \get_object_taxonomies( $post->post_type );
+		// Several plugins just add support to post-formats but don't register post_format taxonomy.
+		if ( \post_type_supports( $post->post_type, 'post-formats' ) && ! in_array( 'post_format', $post_taxonomies, true ) ) {
+			$post_taxonomies[] = 'post_format';
+		}
+
+		$taxonomies_excludelist = $options['taxonomies_excludelist'];
+		if ( ! is_array( $taxonomies_excludelist ) ) {
+			$taxonomies_excludelist = [];
+		}
+
+		if ( ! $options['copy_format'] ) {
+			$taxonomies_excludelist[] = 'post_format';
+		}
+
+		if ( $options['use_filters'] ) {
+			/**
+			 * Filters the taxonomy excludelist when copying a post.
+			 *
+			 * @param array $taxonomies_excludelist The taxonomy excludelist from the options.
+			 *
+			 * @return array
+			 */
+			$taxonomies_excludelist = \apply_filters( 'duplicate_post_taxonomies_excludelist_filter', $taxonomies_excludelist );
+		}
+
+		$post_taxonomies = array_diff( $post_taxonomies, $taxonomies_excludelist );
```

Comment:
array_diff should be backslashed

Reply:
See above, being a standard PHP function...